### PR TITLE
Makes framework universal

### PR DESCRIPTION
**Pull request by izqui**
Previous PR #344, created another target that resulted in the framework having another name. 

This was a problem because when using CocoaPods the name of the module would be `CocoaAsyncSocket` and when adding with Carthage `CocoaAsyncSocketMac`. 

It has been solved now.

### CODE REVIEW

**Comment by jeeftor** [comment]
Universal frameworks (i think aren't officially supported at this time) i believe.

**Comment by izqui** [comment]
They are, the problem with my PR is that the checks are failing because I forgot to update the travis file with the new Schema name

**Comment by izqui** [comment]
Travis test should be passing now

**Comment by chrisballinger** [comment]
This method seems preferable, if you can fix the Travis check I'll merge this

**Comment by izqui** [comment]
Great. Fixed it now @chrisballinger!

**Comment by jeeftor** [comment]
https://github.com/Carthage/Carthage/issues/340

**Comment by chrisballinger** [comment]
@jeeftor Can you explain what you meant by this link?

**Comment by jeeftor** [comment]
I'm in the car. I'll check later. Universal framework is preferred. But there are a few notes on Carthage that they won't support universal because not officially supported by apple. If you can Carthage two targets from this build with Carthage it's great. But not sure it will let you do that.

File: CocoaAsyncSocket.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -226,8 +226,10 @@
 				IPHONEOS_DEPLOYMENT_TARGET = 9.0;
 				MTL_ENABLE_DEBUG_INFO = YES;
 				ONLY_ACTIVE_ARCH = YES;
-				SDKROOT = iphoneos;
+				SDKROOT = macosx10.11;
+				SUPPORTED_PLATFORMS = "iphonesimulator iphoneos macosx";
 				TARGETED_DEVICE_FAMILY = "1,2";
+				VALID_ARCHS = "i386 x86_64 arm64 amr7s arm7";
```
Comment (chrisballinger) [inline CocoaAsyncSocket.xcodeproj/project.pbxproj]: @izqui `amr7s`? The correct values for the last two should be `armv7` and `armv7s`.

File: CocoaAsyncSocket.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -266,9 +268,11 @@
 				GCC_WARN_UNUSED_VARIABLE = YES;
 				IPHONEOS_DEPLOYMENT_TARGET = 9.0;
 				MTL_ENABLE_DEBUG_INFO = NO;
-				SDKROOT = iphoneos;
+				SDKROOT = macosx10.11;
+				SUPPORTED_PLATFORMS = "iphonesimulator iphoneos macosx";
 				TARGETED_DEVICE_FAMILY = "1,2";
 				VALIDATE_PRODUCT = YES;
+				VALID_ARCHS = "i386 x86_64 arm64 amr7s arm7";
```
Comment (chrisballinger) [inline CocoaAsyncSocket.xcodeproj/project.pbxproj]: @izqui Same thing down here

**Comment by izqui** [comment]
Huh, @jeeftor is right. I'm having trouble installing via carthage for iOS. I hadn't tested it.

Will look more into it tomorrow.

File: CocoaAsyncSocket.xcodeproj/project.pbxproj (hunk 2 above)
Comment (izqui) [inline CocoaAsyncSocket.xcodeproj/project.pbxproj]: ups, that's a typo, sorry

Comment (jeeftor) [inline CocoaAsyncSocket.xcodeproj/project.pbxproj]: So if you do cathafe update do you get both build/iOS and build/Mac
?

If so then everything should be good.

**Comment by jeeftor** [comment]
That's why I put a different pull request in. Two targets. Same framework name but different target name. It's the same as how Alamo fire does it.

**Comment by izqui** [comment]
No it doesn't work, it is only building for Mac. I hadn't tested because it assumed it would break iOS support. Sorry.

**Comment by jeeftor** [comment]
Ok now that i'm not in the car.

Carthage says they **WILL NOT** support single target universal framework: as per https://github.com/Carthage/Carthage/issues/340#issuecomment-74725718

> We might support this in the future, but I haven't seen any evidence that Apple is really interested in supporting this case.

This is why I made PR #346 which by far is an inferior solution because you are stuck with two targets, but it does work and you can build both targets.  

If you make a `Cartfile` with the line

`github "jeeftor/CocoaAsyncSocket" "master"`

and you do

`carthage update`

you get:

`/Carthage/Build/Mac/CocoaAsyncSocket.framework`
`/Carthage/Build/iOS/CocoaAsyncSocket.framework`

**Comment by chrisballinger** [comment]
Two targets is fine as long as the module name is the same
`CocoaAsyncSocket`

**Comment by jeeftor** [comment]
It looks like possibly version 0.9.0 of carthage will support this - might want to revisit it in a day or two.

**Pull request closed without merging**